pandora-client-web/src/graphics/useTexture.ts

```diff
@@ -3,50 +3,49 @@ import { useEffect, useRef, useState } from 'react';
 import { GraphicsManagerInstance } from '../assets/graphicsManager';
 import { useObservable } from '../observable';
 
+const RESULT_NO_TEXTURE = {
+	image: '',
+	texture: Texture.EMPTY,
+} as const;
+
+const SPECIAL_TEXTURES: ReadonlyMap<string, Texture> = new Map<string, Texture>([
+	['', Texture.EMPTY],
+	['*', Texture.WHITE],
+]);
+
 /**
  * Resolves an image string to a texture. Supported image formats:
  * - `''` (empty string): `Texture.EMPTY`
  * - `'*'`: `Texture.WHITE`
  * - `data:image/*;base64,...`: The image created from the data blob
  * - `https://...`: Image available on URL
  * @param image - The image to resolve to texture
- * @param preferBlank - True: prefer blank (`Texture.EMPTY`) when texture is not ready; False: Reuse last ready texture when new texture is not ready
+ * @param preferBlank - True: prefer blank (`Texture.EMPTY`) when texture is not ready; False (default): Reuse last ready texture when new texture is not ready
  * @param customGetTexture - Optional getter for resolving http-based textures (is not used for special ones)
  * @returns The requested texture, or `Texture.EMPTY` when the texture is not yet ready
  */
 export function useTexture(
 	image: string | '' | '*',
 	preferBlank: boolean = false,
-	customGetTexture?: (path: string) => Promise<Texture>,
+	customGetTexture?: (path: string) => Texture,
 ): Texture {
 	const manager = useObservable(GraphicsManagerInstance);
 
 	const wanted = useRef('');
-	const wantedGetTexture = useRef<((path: string) => Promise<Texture>) | undefined>(undefined);
-	const [texture, setTexture] = useState({
-		image: '',
-		texture: Texture.EMPTY,
-	});
+	const [texture, setTexture] = useState<{
+		readonly image: string;
+		readonly texture: Texture;
+	}>(RESULT_NO_TEXTURE);
 
 	useEffect(() => {
-		const getTexture = customGetTexture ?? manager?.loader.getTexture.bind(manager.loader);
-		wanted.current = image;
-		wantedGetTexture.current = getTexture;
-		if (!getTexture || image === '') {
-			setTexture({
-				image,
-				texture: Texture.EMPTY,
-			});
+		const loader = manager?.loader;
+		if (customGetTexture || loader == null || SPECIAL_TEXTURES.has(image)) {
+			wanted.current = '';
+			setTexture(RESULT_NO_TEXTURE);
```

Comment:
how does this return `'*', Texture.WHITE`?

Reply:
I split that logic into more generic `SPECIAL_TEXTURES` map, which is then used outside of this async getter to avoid waiting a frame to load these special textures (as the effect is async, it wouldn't resolve until next frame).
The code that handles these overrides is on lines 79-81.